rest-service/manager_rest/rest/resources_v3_1/executions.py

```diff
@@ -13,12 +13,122 @@
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
 #
+from datetime import datetime, timedelta
+
+from flask import request
+from flask_restful.inputs import boolean
+from flask_restful.reqparse import Argument
+
+from cloudify._compat import text_type
+from cloudify.models_states import ExecutionState
 
 from manager_rest.security import SecuredResource
 from manager_rest.security.authorization import authorize
+from manager_rest.rest import resources_v2, rest_decorators
 from manager_rest.resource_manager import get_resource_manager
-from manager_rest.storage import (models,
-                                  get_storage_manager)
+from manager_rest.rest.rest_utils import (get_args_and_verify_arguments,
+                                          get_json_and_verify_params)
+from manager_rest.storage import models,  get_storage_manager, ListResult
+
+
+class Executions(resources_v2.Executions):
+    @authorize('execution_delete')
+    @rest_decorators.marshal_with(models.Execution)
+    @rest_decorators.paginate
+    def delete(self, pagination=None):
+        request_dict = get_json_and_verify_params({
+            'keep_last': {'optional': True, 'type': int},
+            'keep_days': {'optional': True, 'type': int},
+            'keep_since': {'optional': True}
```

Comment:
probably no need for two date filters

i suggest being consistent with the "events delete" endpoint, and using the same mechanisms - please take a look how events delete takes different arguments in the CLI, then formats them to a single `_range` in the rest-client, and then in the restservice, it just uses `@rest_decorators.rangeable()`